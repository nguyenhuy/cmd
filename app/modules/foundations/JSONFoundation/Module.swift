Target.module(
  name: "JSONFoundation",
  dependencies: [],
  testDependencies: [])
