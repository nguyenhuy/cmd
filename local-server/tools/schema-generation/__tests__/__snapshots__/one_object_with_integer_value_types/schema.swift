// swiftformat:disable all
// This file is generated from ./local-server/src/server/schemas/schemaFile.ts by `yarn export-schema-swift`.
// Do not edit it manually.

import Foundation
import JSONFoundation

extension Schema {
  public struct ValueType: Codable, Sendable {
    public let line: Int
    public let column: Int?
  
    private enum CodingKeys: String, CodingKey {
      case line = "line"
      case column = "column"
    }
  
    public init(
        line: Int,
        column: Int? = nil
    ) {
      self.line = line
      self.column = column
    }
  
    public init(from decoder: Decoder) throws {
      let container = try decoder.container(keyedBy: CodingKeys.self)
      line = try container.decode(Int.self, forKey: .line)
      column = try container.decodeIfPresent(Int?.self, forKey: .column)
    }
  
    public func encode(to encoder: Encoder) throws {
      var container = encoder.container(keyedBy: CodingKeys.self)
      try container.encode(line, forKey: .line)
      try container.encodeIfPresent(column, forKey: .column)
    }
  }}
