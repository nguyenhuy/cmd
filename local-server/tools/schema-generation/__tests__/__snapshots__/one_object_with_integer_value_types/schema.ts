export interface ValueType {
  /**
   * @format integer
   */
  line: number;
  /**
   * @format integer
   */
  column?: number;
}
