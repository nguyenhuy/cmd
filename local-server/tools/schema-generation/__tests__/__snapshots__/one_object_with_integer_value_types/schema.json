{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/ValueType",
  "definitions": {
    "ValueType": {
      "type": "object",
      "properties": {
        "line": {
          "type": "number",
          "format": "integer"
        },
        "column": {
          "type": "number",
          "format": "integer"
        }
      },
      "required": [
        "line"
      ],
      "additionalProperties": false
    }
  }
}